steps/terraform-precheck.yaml

```diff
@@ -0,0 +1,65 @@
+parameters:
+  - name: adolibrary
+    default: cnp-azuredevops-libraries
+
+  - name: environment
+    default: sandbox
+
+  - name: keyvaultName
+    default: ''
+
+  - name: keyvaultSecret
+    default: ''
+
+  - name: location
+    default: 'UK South'
+    values:
+      - 'UK South'
+      - 'UK West'
+
+  - name: serviceConnection
+    default: ''
+
+  - name: tfversion
+    default: ''
```

Comment:
we should remove this as a parameter and detect the version with tfenv
that way we don't have to maintain this in the pipeline and in the codebase, as they often get out of sync

https://github.com/tfutils/tfenv#min-required

Reply:
agreed and will look into updating to use tfenv.